sdk/core/core/inc/az_json.h

```diff
@@ -106,44 +120,44 @@ AZ_NODISCARD AZ_INLINE az_json_token az_json_token_string(az_span value)
 }
 
 /*
- * @brief az_json_token_string returns a az_json_token containing an object.
+ * @brief az_json_token_string returns an az_json_token containing an object.
  *
  * @param value A span over an object indicating how the az_json_token should be initialized.
  */
 AZ_NODISCARD AZ_INLINE az_json_token az_json_token_object(az_span value)
 {
   return (az_json_token){
-    .kind = AZ_JSON_TOKEN_OBJECT,
+    .kind = AZ_JSON_TOKEN_OBJECT_START,
```

Comment:
Given my intent is to get some reasonable break down of the changes, I wanted to minimize the changes to other APIs like token, parser, etc. Since this code is going to change more anyway, I would like to keep this as is for now and address later. It is possible we remove az_json_token or change its shape significantly.